src/pages/Visualization/components/LeftSection/components/CodeEditor.tsx

```diff
@@ -29,35 +42,178 @@ const CodeEditor = () => {
   }, [stepIdx, highlightLines]);
 
   useEffect(() => {
-    if (errorLine) {
+    if (editorRef.current && errorLine) {
       displayErrorLine(errorLine);
+      handleEditorDidMount(editorRef.current, monaco);
     }
   }, [errorLine]);
-
+  const handleResetEditor = () => {
+    resetEditor();
+    setGptPin(false);
+    setIsGptToggle(false);
+    if (editorRef.current) {
+      // 데코레이션 컬렉션 리셋
+      if (decorationsCollectionRef.current) {
+        decorationsCollectionRef.current.clear();
+      }
+    }
+  };
   const displayErrorLine = (errorLine: { lineNumber: number; message: string } | null) => {
-    if (!editorRef.current) return;
+    if (!editorRef.current || !errorLine) return;
+
+    const editor = editorRef.current;
+    const model = editor.getModel();
+    if (!model) return;
 
-    const newDecorationIds = editorRef.current.deltaDecorations(decorationIds, [
+    const { lineNumber } = errorLine;
+
+    const endColumn = model.getLineMaxColumn(lineNumber);
+
+    const lineContent = model.getLineContent(lineNumber);
+    let startColumn = 1;
+    for (let i of lineContent) {
+      if (i === " ") {
+        startColumn++;
+      } else {
+        break;
+      }
```

Comment:
이런것도 굳이 else가 필요없을 것 같은데 어떻게 생각하세요? 원래 코드 스타일 인건가요?
```suggestion
      if (i !== " ") {
        break;
      }
      startColumn++;
```

Reply:
빈 문자열이 아니면 startColum을 증가시키는걸 그만하고 for문을 나가야해서 else가 필요합니다

Reply:
빈 문자열이 아니면 break를 실행하니 똑같은 코드 아닌가요? if와 else 위치만 바꾼건데